tests/rptest/tests/audit_log_tests.py

```diff
@@ -155,25 +563,378 @@ def combine(acc, x):
+                                                            self.password,
+                                                            self.algorithm)),
+            log_config=LoggingConfig('info',
+                                     logger_levels={
+                                         'auditing': 'trace',
+                                         'kafka': 'trace',
+                                         'security': 'trace'
+                                     }))
+
+    def setup_cluster(self):
+        self.admin.create_user(self.username, self.password, self.algorithm)
+        self.super_rpk.sasl_allow_principal(
+            principal=self.username,
+            operations=['all'],
+            resource='topic',
+            resource_name="*",
+            username=self.redpanda.SUPERUSER_CREDENTIALS[0],
+            password=self.redpanda.SUPERUSER_CREDENTIALS[1],
+            mechanism=self.redpanda.SUPERUSER_CREDENTIALS[2])
+
+    @staticmethod
+    def authn_filter_function(service_name, username: str, protocol_id: int,
+                              protocol_name: Optional[str], record):
+        return record['class_uid'] == 3002 and record['service'][
+            'name'] == service_name and record['user'][
+                'name'] == username and record[
+                    'auth_protocol_id'] == protocol_id and (
+                        protocol_name is not None and record['auth_protocol']
+                        == protocol_name) and record['status_id'] == 1
+
+    @staticmethod
+    def authn_failure_filter_function(service_name, username: str,
+                                      protocol_id: int,
+                                      protocol_name: Optional[str],
+                                      error_msg: str, record):
+        return record['class_uid'] == 3002 and record['service'][
+            'name'] == service_name and record['user'][
+                'name'] == username and record[
+                    'auth_protocol_id'] == protocol_id and (
+                        protocol_name is not None
+                        and record['auth_protocol'] == protocol_name
+                    ) and record['status_id'] == 2 and record[
+                        'status_detail'] == error_msg
+
+    @cluster(num_nodes=4)
+    def test_authn_messages(self):
+        """Verifies that authentication messages are audited
+        """
+        self.setup_cluster()
+
+        # Now attempt to get the topic list as the regular user
+        user_rpk = self.get_rpk()
+
+        _ = user_rpk.list_topics()
+
+        records = self.read_all_from_audit_log(
+            partial(self.authn_filter_function, self.kafka_rpc_service_name,
+                    self.username, 99, "SASL-SCRAM"),
+            lambda records: self.aggregate_count(records) >= 1)
+
+        assert len(
+            records) == 1, f"Expected only one record got {len(records)}"
+
+        [validate_audit_schema(record) for record in records]
```

Comment:
I'm thinking maybe we should move this into `read_all_from_audit_log` so that whenever any test initiates a read  from the audit log we are asserting by default that its a proper OCSF message

Reply:
Thats a good idea.